src/payment/payment-action-creator.spec.ts

```diff
@@ -249,5 +256,41 @@ describe('PaymentActionCreator', () => {
                 },
             ]);
         });
+
+        it('passes "set_as_default_stored_instrument" flag as null to the paymentRequestTransformer when vaulting, but not as default', () => {
+            const payment = getPayment();
+            const { methodId, gatewayId } = payment;
+
+            paymentActionCreator.initializeOffsitePayment({ methodId, gatewayId, shouldSaveInstrument: true })(store);
+
+            expect(paymentRequestTransformer.transform).toHaveBeenCalledWith(
+                expect.objectContaining({
+                    paymentData: expect.objectContaining({
+                        formattedPayload: expect.objectContaining({
+                            set_as_default_stored_instrument: null,
+                        }),
+                    }),
+                }),
+                expect.anything()
+            );
+        });
+
+        it('passes "set_as_default_stored_instrument" flag as true to the paymentRequestTransformer when vaulting and making default', () => {
+            const payment = getPayment();
+            const { methodId, gatewayId } = payment;
+
+            paymentActionCreator.initializeOffsitePayment({ methodId, gatewayId, shouldSaveInstrument: true, setAsDefaultInstrument: true })(store);
```

Comment:
Just wondering, can `setAsDefaultInstrument` be passed without `shouldSaveInstrument` ?

Reply:
Generally speaking, it can also be passed when when paying with an already vaulted, but non-defaulted, instrument